simd/src/Kokkos_SIMD_Common.hpp

```diff
@@ -31,7 +31,17 @@ class simd;
 template <class T, class Abi>
 class simd_mask;
 
-struct element_aligned_tag {};
+class element_aligned {};
+class vector_aligned {};
+
+template <typename... Flags>
+struct loadstore_flags {};
+
+inline constexpr loadstore_flags<element_aligned> loadstore_default{};
+inline constexpr loadstore_flags<vector_aligned> loadstore_aligned{};
+
+using element_aligned_tag = loadstore_flags<element_aligned>;
+using vector_aligned_tag  = loadstore_flags<vector_aligned>;
```

Comment:
https://en.cppreference.com/w/cpp/experimental/simd/element_aligned has
```
struct element_aligned_tag {};
inline constexpr element_aligned_tag element_aligned{};
```
and `element_aligned` is a `value`, not a `type`. 
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1928r3.pdf has
```
struct element_aligned_tag {};
struct vector_aligned_tag {};
template<size_t> struct overaligned_tag {};
inline constexpr element_aligned_tag element_aligned{}
```
Can you clarify?

Reply:
I think it is my mistake to name these classes `element_aligned` and `vector_aligned`. I'll rename these to something else.

In [p1928r4](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1928r4.pdf), it has a section (4.14) about renaming alignment flags from `[element|vector|*]_aligned` to loadstore_flags (`loadstore_[default|aligned|*]`). 
```
inline constexpr loadstore_flags<see below> loadstore_convert{};
inline constexpr loadstore_flags<see below> loadstore_aligned{};
template<size_t N> requires (has_single_bit<N>)
inline constexpr loadstore_flags<see below> loadstore_overaligned{};
...
```
But for backward compatibility, `element_aligned_tag` is left here as an alias for the new loadstore_flag.

[p1928r6](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1928r6.pdf), however, has another line in the changelog about further renaming these `loadstore_flags` to `simd_flags`. Although I can't seem to find any note on this topic in [p1928r5](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1928r5.pdf), the synopsis provides following:
```
template <class... Flags> struct simd_flags;
inline constexpr simd_flags<> simd_flag_default{};
inline constexpr simd_flags<see below> simd_flag_convert{};
inline constexpr simd_flags<see below> simd_flag_aligned{};
template<size_t N> requires (has_single_bit(N))
inline constexpr simd_flags<see below> simd_flag_overaligned{};
```
So I'll rename `loadstore_flags` to `simd_flags` as well.

Reply:
Removed `element_aligned` class as it's no longer needed with the latest changes and renamed `vector_aligned` to `simd_alignment_vector_aligned`. I am open to suggestions for more suitable name for this class.
Also replaced `loadstore_flags` to `simd_flags`.
See af28107.

Reply:
https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1928r7.pdf also has
```
template <class... Flags> struct simd_flags;
inline constexpr simd_flags<> simd_flag_default{};
inline constexpr simd_flags<see below > simd_flag_convert{};
inline constexpr simd_flags<see below > simd_flag_aligned{};
template<size_t N> requires (has_single_bit(N))
inline constexpr simd_flags<see below > simd_flag_overaligned{};
```